### PR TITLE
Table - split HTML markup into React components

### PR DESCRIPTION
**Pull request by lordpixel**
This PR separates the previous fixture into React components. It also adds an `Export` fixture for demonstrating `bulk actions`.

### CODE REVIEW

File: packages/cascara/src/ui/Table/Table.js
Comment (brianespinosa) [inline packages/cascara/src/ui/Table/Table.js:11]: It looks like right now all the data is getting defined directly in the table context? How are end users getting data into that context? I don't think we want to export contexts from Cascara. We should probably be exposing props on this Table and then passing that into context within this component. Unless this Table is not actually the official component where that happens?

File: packages/cascara/src/ui/Table/TableHeader.js
Comment (brianespinosa) [inline packages/cascara/src/ui/Table/TableHeader.js:10]: I think we should actually make TableHeader and TableFooter separate components entirely. The reason for this is eventually the styles are going to be tied to these two areas directly with modules. And the logic inside them may be different. 

- [ ] Maybe set up a separate TableFooter component?

File: packages/cascara/src/ui/Table/fixtures/TableStyleTest.module.scss
Comment (brianespinosa) [inline packages/cascara/src/ui/Table/fixtures/TableStyleTest.module.scss:1]: This is not how the styles are actually going to be attached to all of these table elements. This was initially just for that test. We can still leave this way for now... but we actually want to update these to apply them directly as CSS Modules to each of the table elements.

These items should probably also get renamed with classes to set those styles accordingly. That can be a separate task though: https://espressive.atlassian.net/browse/DEV-12472

File: packages/cascara/src/ui/Table/index.js
Comment (brianespinosa) [inline packages/cascara/src/ui/Table/index.js:10]: I see... we are using the index.js to set the parameters.

This might present some issues for us in terms of how the exporting and naming of the components work. For instance, our doc typing I do not think will work with this if we define `index.js` as the file to run typing on. The docgen I think will also fail because I do not think exporting default from an index file will apply a component name. 

Maybe this will be okay if we define a component name? I think the problem with that is we are defining the index as Table and then we also have a component called Table. This is likely to be a confusing pattern for devs. 

- [ ] Let's talk about this together and figure out what the pros and cons of this pattern are.

**Review by brianespinosa — CHANGES_REQUESTED** [review]

**Comment by brianespinosa** [comment]
@lordpixel I made a task on all items that I _think_ need to be resolved on this PR. Anything that can be done separately, I went ahead and made a Jira task for it so it does not block this PR.

File: .vscode/settings.json
Comment (lordpixel) [reply .vscode/settings.json:41]: This change was generate from `esp-app-web` somehow, let me put it back.

File: packages/cascara/package.json
Comment (lordpixel) [reply packages/cascara/package.json:26]: this is a dumb mistake on my part, it was not my intention to add it to this project but somehow I did. Thanks for the catch!

File: packages/cascara/src/index.js
Comment (lordpixel) [reply packages/cascara/src/index.js:3]: agreed, will move this to modules once it's good to go.

File: packages/cascara/src/ui/Table/ActionBar.js
Comment (lordpixel) [reply packages/cascara/src/ui/Table/ActionBar.js:10]: Yup, it is used in both scopes.

File: packages/cascara/src/ui/Table/index.js
Comment (lordpixel) [reply packages/cascara/src/ui/Table/index.js:10]: yup this is not a good pattern, I didn't want to spend time on fixing it because it was just a fixture.
Let me finish the convertion to React components and fix this, thanks for the catch!

File: packages/cascara/src/ui/Table/Table.js
Comment (lordpixel) [reply packages/cascara/src/ui/Table/Table.js:11]: You are right, Table is wrapped by the TableContext. The index in `packages/cascara/src/ui/Table` exports the

File: packages/cascara/src/ui/Table/TableHeader.js
Comment (lordpixel) [reply packages/cascara/src/ui/Table/TableHeader.js:10]: sure thing, we will have those as separate components.

File: .vscode/settings.json
Comment (brianespinosa) [reply .vscode/settings.json:41]: @lordpixel you put these back into the extensions.json file. The extensions.json file controls what extensions VSCode will suggest to people when opening the editor.

- [ ] Add settings back here

File: .vscode/extensions.json
Comment (brianespinosa) [inline .vscode/extensions.json:12]: - [ ]  Please restore the extensions.json file

**Review by brianespinosa — CHANGES_REQUESTED** [review]

File: .vscode/extensions.json
Comment (lordpixel) [reply .vscode/extensions.json]: This is caused by vscode.

File: packages/cascara/src/ui/Table/index.js
Comment (lordpixel) [reply packages/cascara/src/ui/Table/index.js]: The change has been applied, Table is the context container and the index only exports table.

**Review by brianespinosa — APPROVED** [review]

**Comment by brianespinosa** [comment]
@lordpixel there are a few conflicts with develop. As soon as those get resolved, this is officially approved to merge.

It looks like there was a change to that test CSS file that was adding body padding to ALL fixtures (another reason we do not want to add global type styles. ;)

**Review by brianespinosa — APPROVED** [review]

**Pull request merged**
